dotnet/src/Microsoft.Agents.AI.Abstractions.CosmosNoSql/CosmosChatMessageStore.cs

```diff
@@ -0,0 +1,853 @@
+
+    /// <summary>
+    /// Gets the conversation ID associated with this message store.
+    /// </summary>
+    public string ConversationId => this._conversationId;
+
+    /// <summary>
+    /// Gets the database ID associated with this message store.
+    /// </summary>
+    public string DatabaseId => this._databaseId;
+
+    /// <summary>
+    /// Gets the container ID associated with this message store.
+    /// </summary>
+    public string ContainerId => this._containerId;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CosmosChatMessageStore"/> class using a connection string.
+    /// </summary>
+    /// <param name="connectionString">The Cosmos DB connection string.</param>
+    /// <param name="databaseId">The identifier of the Cosmos DB database.</param>
+    /// <param name="containerId">The identifier of the Cosmos DB container.</param>
+    /// <exception cref="ArgumentNullException">Thrown when any required parameter is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when any string parameter is null or whitespace.</exception>
+    public CosmosChatMessageStore(string connectionString, string databaseId, string containerId)
+        : this(connectionString, databaseId, containerId, Guid.NewGuid().ToString("N"))
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CosmosChatMessageStore"/> class using a connection string.
+    /// </summary>
+    /// <param name="connectionString">The Cosmos DB connection string.</param>
+    /// <param name="databaseId">The identifier of the Cosmos DB database.</param>
+    /// <param name="containerId">The identifier of the Cosmos DB container.</param>
+    /// <param name="conversationId">The unique identifier for this conversation thread.</param>
+    /// <exception cref="ArgumentNullException">Thrown when any required parameter is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when any string parameter is null or whitespace.</exception>
+    public CosmosChatMessageStore(string connectionString, string databaseId, string containerId, string conversationId)
+    {
+        if (string.IsNullOrWhiteSpace(connectionString))
+        {
+            throw new ArgumentException("Connection string cannot be null or whitespace.", nameof(connectionString));
+        }
+
+        if (string.IsNullOrWhiteSpace(databaseId))
+        {
+            throw new ArgumentException("Database ID cannot be null or whitespace.", nameof(databaseId));
+        }
+
+        if (string.IsNullOrWhiteSpace(containerId))
+        {
+            throw new ArgumentException("Container ID cannot be null or whitespace.", nameof(containerId));
+        }
+
+        if (string.IsNullOrWhiteSpace(conversationId))
+        {
+            throw new ArgumentException("Conversation ID cannot be null or whitespace.", nameof(conversationId));
+        }
+
+        this._cosmosClient = new CosmosClient(connectionString);
```

Comment:
We have some helpers in the code base that allows in-lining the param validation:
```suggestion
        this._cosmosClient = new CosmosClient(Throw.IfNullOrWhitespace(connectionString));
```

Looks like you are already including them via `<InjectSharedThrow>true</InjectSharedThrow>` in the project file.